pkg/policy/rule.go

```diff
@@ -27,6 +29,19 @@ import (
 
 type rule struct {
 	api.Rule
+
+	// mutex protects everything below it (i.e., not the rule itself).
+	mutex lock.RWMutex
+
+	// localRuleConsumers is the set of the numeric identifiers which this rule
+	// selects which are node-local (e.g., Endpoint).
+	localRuleConsumers map[uint16]*identity.Identity
+
+	// processedConsumers tracks which consumers have been 'processed' - that is,
+	// it determines whether the consumer has actually been processed in relation
+	// to this rule. It does *not* encode whether the rule selects the consumer;
+	// that is what localRuleConsumers is for.
+	processedConsumers map[uint16]struct{}
```

Comment:
1. Rename this.
2. Group with localRuleConsumers in a single type to signify that they are 'closed' and related.